src/Sql/Sql/VulnerabilityAssessment/Cmdlet/VulnerabilityAssessmentSettings/Server/ClearSqlServerVulnerabilityAssessmentSettings.cs

```diff
@@ -0,0 +1,102 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System;
+using Microsoft.Azure.Commands.Sql.VulnerabilityAssessment.Model;
+using System.Management.Automation;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Commands.Sql.Server.Model;
+
+namespace Microsoft.Azure.Commands.Sql.VulnerabilityAssessmentSettings.Cmdlet
+{
+    /// <summary>
+    /// Clears the Vulnerability Assessment on a specific server.
+    /// </summary>
+    [Cmdlet("Clear", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "SqlServerVulnerabilityAssessmentSettings", DefaultParameterSetName = ClearByNameParameterSet, SupportsShouldProcess = true), OutputType(typeof(ServerVulnerabilityAssessmentSettingsModel))]
+    public class ClearSqlServerVulnerabilityAssessmentSettings : ClearSqlVulnerabilityAssessmentSettingsBase
+    {
```

Comment:
It is handled in the base class - AzureRMCmdlet

Reply:
Could you please link me to the handler? I'm looking for something like this: 
```Powershell
[Cmdlet(VerbsCommon.Remove, "Foo", SupportsShouldProcess = true)]
public class RemoveFoo
{
    ...
    public override void ExecuteCmdlet()
    {
        if (ShouldProcess(target, actionMessage))
        {
            // make changes
        }
    }
}
```

Reply:
The handler is implemented in assembly Microsoft.Azure.PowerShell.Common
in class AzurePSCmdlet:

```powershell
protected virtual void ConfirmAction(string processMessage, string target, Action action)
{
	if (_qosEvent != null)
	{
		_qosEvent.PauseQoSTimer();
	}
	if (base.ShouldProcess(target, processMessage))
	{
		if (_qosEvent != null)
		{
			_qosEvent.ResumeQosTimer();
		}
		action();
	}
}
```


(class AzureRMCmdlet calls method ConfirmAction)

Reply:
@talhers  The `ConfirmAction` is a [helper method](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/should-process-confirm-impact.md#additional-common-methods) that you may use to properly implement confirmation in the implementation of ProcessRecord()..

Only your cmdlet knows  which action should be executed in case of confirmation or `-Whatif`.

Please take a look at how the helper method is used in a random cmdlet from our repo: https://github.com/Azure/azure-powershell/blob/master/src/Network/Network/ApplicationGateway/RemoveAzureApplicationGatewayCommand.cs#L56

Reply:
In Sql module, all cmdlet classes inherit from AzureSqlCmdletBase, and implement PersistChanges method.
PersistChanges is called from [AzureSqlCmdletBase.ExecuteCmdlet](https://github.com/Azure/azure-powershell/blob/master/src/Sql/Sql/Common/AzureSqlCmdletBase.cs#L119
) if ConfirmAction returns true.


For example, cmdlet [NewAzureSqlDatabase ](https://github.com/Azure/azure-powershell/blob/master/src/Sql/Sql/Database/Cmdlet/NewAzureSqlDatabase.cs)- uses "SupportsShouldProcess = true", but doesn't directly use ConfirmAction method, because it is used in ExecuteCmdlet in base class AzureSqlCmdletBase.

Reply:
Now I see. Thanks.